receiver/rabbitmqreceiver/metadata.yaml

```diff
@@ -1 +1,65 @@
 name: rabbitmqreceiver
+
+attributes:
+  rabbitmq.queue.name:
+    description: The name of the RabbitMQ queue.
+  rabbitmq.node.name:
+    description: The name of the RabbitMQ node.
+  rabbitmq.vhost.name:
+    description: The name of the RabbitMQ vHost.
```

Comment:
Why do we need attributes to be prefixed with `rabbitmq`? Maybe can be just `queue`, `node` and `vhost`?

Reply:
We don't have to it's just a convention I saw in some other scrapers so I followed it. Saw it in [Elasticsearch](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/elasticsearchreceiver/metadata.yaml#L4-L7) and [CouchDB](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/couchdbreceiver/metadata.yaml#L4-L5). Two recently added ones.

I can change it if it's not something we want to do.

Reply:
I am to blame for both the Elasticsearch and CouchDB examples of this. Here's why:

Those that are namespaced with the technology are meant to be resource attributes. Those that are just `queue`, `node`, etc are data point attributes. 

In the latter case, we have all the context we need to understand that `queue` refers to a rabbitmq queue, because I'm looking at a metric called `rabbitmq.something`. Resource attributes though, need to be interpreted in a global context, so require the namespace.

That said, the `metadata.yaml` isn't really designed to differentiate between the two, so this is simply a convention I've pushed for, as we continue to hash out the best patterns. We should decide if resource attributes ought to be included in the `metadata.yaml` at all. If so, we should probably differentiate them in the structure somehow.

Reply:
Got it. Thanks for explaining. We do need to do something with the resource attribute. Maybe at least move it to another field. I can submit an issue for that.

Speaking about this change. What about `rabbitmq.queue.name`? Do you think it's a resource attribute?

Reply:
> What about `rabbitmq.queue.name`? Do you think it's a resource attribute?

Intuitively, I wanted to challenge this as well but couldn't pinpoint a clear justification.

The project describes a [Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#resources) as an `entity for which telemetry is recorded`. An [entity](https://www.merriam-webster.com/dictionary/entity) is `something that exists by itself : something that is separate from other things`. In a world of software abstractions, it's not clear to me exactly how we could say that a queue is _not_ a separate thing for which telemetry can be recorded.

I don't know that this is the intended interpretation but it's the clearest heuristic I could identify. 

What do you think?

Reply:
Yes, conceptually it seems valid to put `rabbitmq.queue.name` into resource attributes, maybe not as clear as others.

The last topic I'd like to raise is about usability. Do you think if is it useful to be able to aggregate all the datapoints across `queue`, `node` or `vhost`? It would be impossible to do with the new metrics builder functionality if the are resource attributes. Not sure if the aggregation over different resources captured somewhere in the specification. Maybe the metrics builder would need to support this use case in future and take care of resource metrics.

Reply:
> Do you think if is it useful to be able to aggregate all the datapoints across queue, node or vhost?

Definitely. This is a potential motivation to make these data point attributes but it would seem to come as a tradeoff against the notion of resource identification. Ultimately, I think we have this same issue eventually. e.g. aggregation across resources collected by multiple scraper instances, or even semantically equivalent resources captured by unrelated instrumentation.

> Not sure if the aggregation over different resources captured somewhere in the specification. 

I'm not aware of it. There is a notion of merging resources but this appears to be intended for merging telemetry describing the same resource.

I think there is an open question as to what are the necessary requirements in order for metric aggregation across resources to be well defined. Establishing this would allow for decisions about identification and aggregation to be made without tradeoffs.

> Maybe the metrics builder would need to support this use case in future and take care of resource metrics.

Despite not having clarity on the _necessary_ requirements for metric aggregation across resources, I think we can tentatively claim that the metric builder has _sufficient_ context to perform such aggregations.

Reply:
Ok sounds good. I'll look into the aggregation across resources approach. It doesn't seem like something conceptually wrong.

I'm good with merging this PR

Reply:
Do we plan to also submit semantic conventions PR for rabbitmq resource attributes?

Reply:
I've been trying to follow this discussions I'm under the impression there aren't semantic conventions for resource attributes yet? If so I'll happily add them if someone can point me to them.